File: Java/Program/Product.java
public class Product {
    private static int lastId = 0;
    private int idProduct;
    private String name;
    private String brand;
    private int price;

    public Product(String name, String brand, int price) {
        lastId++;
        this.idProduct = lastId;
        this.name = name;
        this.brand = brand;
        this.price = price;
    }

    public int getIdProduct() {
        return idProduct;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }
}

File: CPP/Program/Product.cpp
#include "Product.h"

int Product::last_id = 0;

Product::Product(std::string name, std::string brand, int price) {
    last_id++;
    id_product = last_id;
    this->name = name;
    this->brand = brand;
    this->price = price;
}

int Product::getIdProduct() {
    return id_product;
}

std::string Product::getName() {
    return name;
}

void Product::setName(std::string name) {
    this->name = name;
}

std::string Product::getBrand() {
    return brand;
}

void Product::setBrand(std::string brand) {
    this->brand = brand;
}

int Product::getPrice() {
    return price;
}

void Product::setPrice(int price) {
    this->price = price;
}
